lib/overcommit/hook_context/pre_push.rb

```diff
@@ -18,17 +18,14 @@ def pushed_refs
     end
 
     def modified_files
-      @modified_files ||= Overcommit::GitRepo.modified_files(refs: ref_range)
+      @modified_files ||= pushed_refs.map(&:modified_files).flatten
```

Comment:
Set my comments below for how you may want to speed this up for large ref ranges containing many large file sets.